sound/soc/sof/sof-acpi-dev.c

```diff
@@ -193,44 +110,20 @@ static int sof_acpi_probe(struct platform_device *pdev)
 
 	return ret;
 }
+EXPORT_SYMBOL_NS(sof_acpi_probe, SND_SOC_SOF_ACPI_DEV);
 
-static int sof_acpi_remove(struct platform_device *pdev)
+int sof_acpi_remove(struct platform_device *pdev)
 {
+	struct device *dev = &pdev->dev;
+
 	if (!(sof_acpi_debug & SOF_ACPI_DISABLE_PM_RUNTIME))
-		pm_runtime_disable(&pdev->dev);
+		pm_runtime_disable(dev);
 
 	/* call sof helper for DSP hardware remove */
-	snd_sof_device_remove(&pdev->dev);
+	snd_sof_device_remove(dev);
 
 	return 0;
 }
-
-#ifdef CONFIG_ACPI
-static const struct acpi_device_id sof_acpi_match[] = {
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_BROADWELL)
-	{ "INT3438", (unsigned long)&sof_acpi_broadwell_desc },
-#endif
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_BAYTRAIL)
-	{ "80860F28", (unsigned long)&sof_acpi_baytrail_desc },
-	{ "808622A8", (unsigned long)&sof_acpi_cherrytrail_desc },
-#endif
-	{ }
-};
-MODULE_DEVICE_TABLE(acpi, sof_acpi_match);
-#endif
-
-/* acpi_driver definition */
-static struct platform_driver snd_sof_acpi_driver = {
-	.probe = sof_acpi_probe,
-	.remove = sof_acpi_remove,
-	.driver = {
-		.name = "sof-audio-acpi",
```

Comment:
Without this, `sof_parent `will be null and non SOF sound card name will be set and driver name will be NULL.

https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/cht_bsw_max98090_ti.c#L602

Reply:
> Without this, `sof_parent` will be 0, non-SOF sound card name will be set and driver name will be NULL.
> 
> https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/cht_bsw_max98090_ti.c#L602

Yeah, `"sof-audio-acpi"` is used by [snd_soc_acpi_sof_parent](https://github.com/thesofproject/linux/blob/topic/sof-dev/include/sound/soc-acpi.h#L174), so we need to modify `snd_soc_acpi_sof_parent` as well.

Reply:
ah, would that explain the card name issue? We should move to a partial match then. thanks for looking into this.

Reply:
> ah, would that explain the card name issue?

Yes, the card name is selected by the return value of `snd_soc_acpi_sof_parent(&pdev->dev);`
```
#define SOF_CARD_NAME "bytcht max98090" /* card name will be 'sof-bytcht max98090 */
#define SOF_DRIVER_NAME "SOF"

#define CARD_NAME "chtmax98090"
#define DRIVER_NAME NULL /* card name will be used for driver name */

...

	sof_parent = snd_soc_acpi_sof_parent(&pdev->dev);

	/* set card and driver name */
	if (sof_parent) {
		snd_soc_card_cht.name = SOF_CARD_NAME;
		snd_soc_card_cht.driver_name = SOF_DRIVER_NAME;
	} else {
		snd_soc_card_cht.name = CARD_NAME;
		snd_soc_card_cht.driver_name = DRIVER_NAME;
	}
```

Reply:
fixed with strncmp now and tested on CHT device. Thanks @fredoh9 for catching this, I didn't think of this hard-coding at all.